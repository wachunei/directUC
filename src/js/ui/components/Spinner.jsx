import React from "react";
import styled, { keyframes } from "styled-components";

function Spinner({ ...props }) {
  return (
    <svg
      width="45"
      height="45"
      viewBox="0 0 45 45"
      stroke="#1F95EB"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M43 22.5C43 33.8218 33.8218 43 22.5 43S2 33.8218 2 22.5 11.1782 2 22.5 2"
        strokeWidth="4"
      />
    </svg>
  );
}

const rotation = keyframes`
  from {
    transform: rotate(0);
  }
  to {
    transform: rotate(360deg);
  }
`;

const RotatingSpinner = styled(Spinner)`
  animation: ${rotation} 1s ease infinite;
`;

export default RotatingSpinner;
