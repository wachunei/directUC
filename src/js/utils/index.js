export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
