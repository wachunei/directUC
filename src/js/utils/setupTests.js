// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
