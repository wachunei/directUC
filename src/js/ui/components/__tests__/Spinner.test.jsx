import Spinner from "../Spinner";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Spinner Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Spinner />);
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with darkTheme", () => {
    const tree = renderTree(<Spinner />, "lightTheme");
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with lightTheme", () => {
    const tree = renderTree(<Spinner />, "darkTheme");
    expect(tree).toMatchSnapshot();
  });
});
