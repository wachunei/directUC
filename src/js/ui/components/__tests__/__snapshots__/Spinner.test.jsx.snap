// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should match snapshot 1`] = `
.c0 {
  -webkit-animation: kJPzGr 1s ease infinite;
  animation: kJPzGr 1s ease infinite;
}

<svg
  class="c0"
  fill="none"
  height="45"
  stroke="#1F95EB"
  viewBox="0 0 45 45"
  width="45"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M43 22.5C43 33.8218 33.8218 43 22.5 43S2 33.8218 2 22.5 11.1782 2 22.5 2"
    stroke-width="4"
  />
</svg>
`;

exports[`Spinner Component should match snapshot with darkTheme 1`] = `
.c0 {
  -webkit-animation: kJPzGr 1s ease infinite;
  animation: kJPzGr 1s ease infinite;
}

<svg
  class="c0"
  fill="none"
  height="45"
  stroke="#1F95EB"
  viewBox="0 0 45 45"
  width="45"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M43 22.5C43 33.8218 33.8218 43 22.5 43S2 33.8218 2 22.5 11.1782 2 22.5 2"
    stroke-width="4"
  />
</svg>
`;

exports[`Spinner Component should match snapshot with lightTheme 1`] = `
.c0 {
  -webkit-animation: kJPzGr 1s ease infinite;
  animation: kJPzGr 1s ease infinite;
}

<svg
  class="c0"
  fill="none"
  height="45"
  stroke="#1F95EB"
  viewBox="0 0 45 45"
  width="45"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M43 22.5C43 33.8218 33.8218 43 22.5 43S2 33.8218 2 22.5 11.1782 2 22.5 2"
    stroke-width="4"
  />
</svg>
`;
